assets/src/scripts/monitoring-location/components/hydrograph/index.js

```diff
@@ -75,117 +66,80 @@ export const attachToNode = function(store,
         .select('.loading-indicator-container')
         .call(drawLoadingIndicator, {showLoadingIndicator: true, sizeClass: 'fa-3x'});
 
-    // Fetch time zone
-    const fetchTimeZonePromise = store.dispatch(timeZoneActions.retrieveIanaTimeZone(latitude, longitude));
-    // Fetch waterwatch flood levels
-    store.dispatch(floodDataActions.retrieveWaterwatchData(siteno));
-    let fetchDataPromise;
-    if (showOnlyGraph) {
-        // Only fetch what is needed
-        if (parameterCode && period) {
-            fetchDataPromise = store.dispatch(ivTimeSeriesDataActions.retrieveCustomTimePeriodIVTimeSeries(siteno, parameterCode, period));
-        } else if (parameterCode && startDT && endDT) {
-            // Don't fetch until time zone is available
-            fetchDataPromise = fetchTimeZonePromise.then(() => {
-                return store.dispatch(ivTimeSeriesDataActions.retrieveUserRequestedIVDataForDateRange(siteno, startDT, endDT, parameterCode));
-            });
+    const fetchDataPromise = store.dispatch(retrieveHydrographData(siteno, {
```

Comment:
This retrieves the data needed to render the hydrograph. The promise is not resolved until all data that is required is fetched. But all of the calls occur immediately.